### ISSUE TITLE
Use empty outer context when checking str.format() calls

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#7479]
This is a follow up for https://github.com/python/mypy/pull/7418

Because mypy is still too eager about outer context and `str.format()` has `object` type for its arguments, passing a call to generic function as an argument to `str.format()` may cause false positives. For example:
```python
'{:.3f}'.format(sum(list_of_floats))
```
Fails with 
```
Incompatible types in string interpolation (expression has type "object", placeholder has type "Union[int, float]")
```
A more rectified test case:
```
[case testFormatCallGenerics]
from typing import TypeVar, Union

T = TypeVar('T')
def foo(x: T) -> Union[T, int]: ...

x: str
y: float

"{:.3f}".format(foo(x))  # E: Incompatible types in string interpolation (expression has type "str", placeholder has type "Union[int, float]")
"{:.3f}".format(foo(y))
[builtins fixtures/primitives.pyi]
```

A potential simple fix is to re-accept the expression in empty context before checking the format specifier type.

**Comment by Architect0rr** [comment]
I have the same problem too
